### PR TITLE
Add org.squidowl.halloy

### PR DESCRIPTION
**Pull request by tarkah**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

---

https://github.com/squidowl/halloy

### CODE REVIEW

**Comment by hfiguiere** [comment]
pr is supposed to be against the branch`new-pr`

File: org.squidowl.halloy.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <content_rating type="oars-1.0">
```
Comment (hfiguiere) [inline org.squidowl.halloy.appdata.xml:3]:
```suggestion
  <content_rating type="oars-1.1">
```

**Comment by tarkah** [comment]
> pr is supposed to be against the branch `new-pr`

Ahh let me reopen, thanks

File: org.squidowl.halloy.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <content_attribute id="social-chat">moderate</content_attribute>
+  </content_rating>
+  <id>org.squidowl.halloy</id>
```
Comment (hfiguiere) [inline org.squidowl.halloy.appdata.xml:6]: Is `squidowl.org` under the control of the developers?

File: org.squidowl.halloy.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            "build-options": {
+                "env": {
+                    "CARGO_HOME": "/run/build/halloy/cargo"
+                }
+            },
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+                "cargo --offline build --release --verbose",
+                "mkdir -p /app/share/icons/hicolor/128x128/apps/",
+                "cp logo.png /app/share/icons/hicolor/128x128/apps/org.squidowl.halloy.png",
```
Comment (hfiguiere) [inline org.squidowl.halloy.json:35]:
```suggestion
                "install -Dm644 logo.png /app/share/icons/hicolor/128x128/apps/org.squidowl.halloy.png",
```

File: org.squidowl.halloy.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                "mkdir -p /app/share/metainfo/",
+                "cp org.squidowl.halloy.appdata.xml /app/share/metainfo/org.squidowl.halloy.appdata.xml",
+                "mkdir -p /app/share/applications/",
+                "cp ./assets/halloy.desktop /app/share/applications/org.squidowl.halloy.desktop",
```
Comment (hfiguiere) [inline org.squidowl.halloy.json:39]:
```suggestion
                "install -Dm644 org.squidowl.halloy.appdata.xml /app/share/metainfo/org.squidowl.halloy.appdata.xml",
                "install -Dm644 ./assets/halloy.desktop /app/share/applications/org.squidowl.halloy.desktop",
```

File: org.squidowl.halloy.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    "runtime-version": "22.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "halloy",
+    "finish-args": [
+        "--share=ipc",
+        "--filesystem=xdg-config",
+        "--filesystem=xdg-data",
```
Comment (hfiguiere) [inline org.squidowl.halloy.json:13]: why do you need access to these?

File: org.squidowl.halloy.appdata.xml
Comment (hfiguiere) [inline org.squidowl.halloy.appdata.xml:1]: Also there is this file usptream and it should be used instead

https://github.com/squidowl/halloy/blob/main/assets/halloy.appdata.xml

Comment (tarkah) [reply org.squidowl.halloy.appdata.xml:1]: I'll look at templating this file :+1:

File: org.squidowl.halloy.json (hunk 5 above)
Comment (tarkah) [reply org.squidowl.halloy.json:13]: Use config and state files are persisted here. We're using `HOST_XDG_*` vars to access the host XDG dirs.

File: org.squidowl.halloy.appdata.xml
Comment (hfiguiere) [reply org.squidowl.halloy.appdata.xml:1]: why templating. it's not flatpak specific

File: org.squidowl.halloy.appdata.xml (hunk 2 above)
Comment (tarkah) [reply org.squidowl.halloy.appdata.xml:6]: Yes, we've recently acquired the domain but we don't have a website up yet.

File: org.squidowl.halloy.json (hunk 5 above)
Comment (hfiguiere) [reply org.squidowl.halloy.json:13]: if you are XDG compliant you don't need either.

Comment (tarkah) [reply org.squidowl.halloy.json:13]: I'm not sure I'm following. We prefer to use the host's XDG paths vs `~/.var/app/..`. Unless you're suggesting something else.

Comment (hfiguiere) [reply org.squidowl.halloy.json:13]: you are not indeed. You don't need to set any of these permissions, which are excessive, for this to work.

Comment (tarkah) [reply org.squidowl.halloy.json:13]: I don't see how this is correct. I just removed those and it prevent us accessing those locations `~/.config/halloy/` / `~/.local/share/halloy` on my system.

We are not looking to force users into managing their dotfiles for this application in the `~/.var` location. I'd much rather respect what their host XDG directories are, which these permissions give us access to.

**Pull request closed without merging**